src/main/java/Duke.java

```diff
@@ -1,10 +1,176 @@
+    }
+
+    public static void handleDelete(String[] inputWords) {
+        int index = Integer.parseInt(inputWords[1]);
+        Task task = list.get(index - 1);
+        list.remove(index-1);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + task.toString());
+        System.out.println(String.format("Now you have %d tasks in the list.", list.size()));
+    }
+
+    public static void handleMark(String[] inputWords) {
+        int index = Integer.parseInt(inputWords[1]);
+        Task task = list.get(index - 1);
+        task.mark();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  " + task.toString());
+    }
+
+    public static void handleUnmark(String[] inputWords) {
+        int index = Integer.parseInt(inputWords[1]);
+        Task task = list.get(index - 1);
+        task.unmark();
+        System.out.println("OK,, I've marked this task as not done yet:");
+        System.out.println("  " + task.toString());
+    }
+
+    public static void addTask(Task newTask) {
+        list.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + newTask.toString());
+        System.out.println(String.format("Now you have %d tasks in the list.", list.size()));
+    }
+
+    public static void handleToDo(String[] inputWords) throws EmptyDescriptionException {
+        checkEmptyDescription(inputWords);
+        String description = inputWords[1];
+        Task newTask = new ToDo(description);
+        addTask(newTask);
+    }
+
+    public static void handleDeadline(String[] inputWords) throws EmptyDescriptionException {
+        checkEmptyDescription(inputWords);
+        String[] splitedString = inputWords[1].split(" /by ");
+        String action = splitedString[0];
+        String date = splitedString[1]; // in yyyy-mm-dd HHMM format
+        LocalDateTime inputDateTime = LocalDateTime.parse(date, inputFormatter);
+        String outputDateTime = inputDateTime.format(outputFormatter);
+        Task newTask = new Deadline(action, outputDateTime);
+        addTask(newTask);
+    }
+
+    public static void handleEvent(String[] inputWords) throws EmptyDescriptionException {
+        checkEmptyDescription(inputWords);
+        String[] splitedString = inputWords[1].split(" /from ");
+        String action = splitedString[0];
+        String duration = splitedString[1];
+        String[] fromTo = duration.split(" /to ");
+        String from = fromTo[0];
+        String to = fromTo[1];
+        Task newTask = new Event(action, from, to);
+        addTask(newTask);
+    }
+    public static void getList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            Task task = list.get(i);
+            System.out.println(i+1 + "." + task.toString());
+        }
+    }
+
+    public static boolean isMark(String word) {
```

Comment:
I think abstracting these was a great choice, improved readability.

Reply:
Thank you!